crates/matrix-sdk-base/src/rooms/normal.rs

```diff
@@ -202,7 +206,13 @@ impl Room {
                 Self::MAX_ENCRYPTED_EVENTS,
             ))),
             roominfo_update_sender,
-        }
+            cached_display_name: Arc::new(SyncRwLock::new(None)),
+        };
+
+        // Refill the display name cache.
+        let _ = room.compute_display_name().await;
```

Comment:
Since we restore all rooms at startup, this has the potential to slow down startup quite badly. I guess it won't in the nominal case where most of your rooms have names but still quite worrisome.

That's the primary reason why I suggested the `RoomInfo` for this cache instead.

Reply:
I think it's a code smell to persist derived/computed state, but oh well.

Reply:
Hmm, but why? How would you otherwise avoid the slowdown? Perhaps a benchmark telling us that the slowdown isn't important would settle the discussion either way?

The only other idea would be a deferred filling out of the computed state. But that sounds worse to me.

Reply:
I've put it in the `RoomInfo` now.